drivers/slipmux/include/slipmux_internal.h

```diff
@@ -0,0 +1,202 @@
+ */
+#define SLIPMUX_END_ESC     (0xdcU)
+/**
+ * @brief   Ends escape sequence, next byte is ESC
+ */
+#define SLIPMUX_ESC_ESC     (0xddU)
+/** @} */
+
+/**
+ * @name    SLIPMUX marker bytes
+ * @see     [SLIPMUX](https://tools.ietf.org/html/draft-bormann-t2trg-slipmux-03#section-4)
+ * @{
+ */
+/**
+ * @brief   Starts a diagnostic frame
+ */
+#define SLIPMUX_STDIO_START (0x0aU)
+/**
+ * @brief   Starts a configuration frame
+ */
+#define SLIPMUX_COAP_START (0xa9U)
+/**
+ * @brief   Starts a IP packet frame
+ */
+#define SLIPMUX_NET_START(byte) (\
+            /* is it an IPv4 packet? */ \
+            (byte >= 0x45 && byte <= 0x4f) || \
+            /* or is it an IPv6 packet? */ \
+            (byte >= 0x60 && byte <= 0x6f) \
+        )
+/** @} */
+
+/**
+ * @brief   Decoder internal state
+ */
+enum {
+    /* Device is in no mode (currently did not receiving any data frame) */
+    SLIPMUX_STATE_NONE = 0,
+    /* Device writes handles data as network device */
+    SLIPMUX_STATE_NET,
+    /* Device writes handles data as network device, next byte is escaped */
+    SLIPMUX_STATE_NET_ESC,
+    /* Device writes received data to stdin */
+    SLIPMUX_STATE_STDIN,
+    /* Device writes received data to stdin, next byte is escaped */
+    SLIPMUX_STATE_STDIN_ESC,
+    /* Device writes received data as CoAP message */
+    SLIPMUX_STATE_COAP,
+    /* Device writes received data as CoAP message, next byte is escaped */
+    SLIPMUX_STATE_COAP_ESC,
+    /* Device is in standby, will wake up when sending data */
+    SLIPMUX_STATE_STANDBY,
+    /* Device is in sleep mode */
+    SLIPMUX_STATE_SLEEP,
+};
+
+/**
+ * @brief   Callback for the UART on receiving data
+ */
+void slipmux_rx_cb(void *arg, uint8_t byte);
+
+/**
+ * @brief   Writes one byte to UART
+ *
+ * @param[in] uart  The UART device to write to.
+ * @param[in] byte  The byte to write to @p uart.
+ */
+static inline void slipmux_write_byte(uart_t uart, uint8_t byte)
+{
+    uart_write(uart, &byte, 1U);
+}
+
+/**
+ * @brief   Write multiple bytes SLIPMUX-escaped to UART
+ *
+ * @param[in] uart  The UART device to write to.
+ * @param[in] data  The bytes to write SLIPMUX-escaped to @p uart.
+ * @param[in] len   Number of bytes in @p data.
+ */
+void slipmux_write_bytes(uart_t uart, const uint8_t *data, size_t len);
+
+/**
+ * @brief   Acquire exclusive access for writing
+ *          Guarded for _stdio or _coap, the idea being that if only _net is
+ *          used, you are most likely to care about performance.
+ */
+static inline void slipmux_lock(slipmux_t *dev)
+{
+    (void) dev;
+#if IS_USED(MODULE_SLIPMUX_STDIO) || IS_USED(MODULE_SLIPMUX_COAP)
+    mutex_lock(&dev->mutex);
+#endif
+}
+
+/**
+ * @brief   Release the exclusive access
+ */
+static inline void slipmux_unlock(slipmux_t *dev)
+{
+    (void) dev;
+#if IS_USED(MODULE_SLIPMUX_STDIO) || IS_USED(MODULE_SLIPMUX_COAP)
+    mutex_unlock(&dev->mutex);
+#endif
+}
+
+#if IS_USED(MODULE_SLIPMUX_COAP)
```

Comment:
```suggestion
#if IS_USED(MODULE_SLIPMUX_COAP) || DOXYGEN
```

if you wanted those to show up in the documentation